grc/core/utils/expr_utils.py

```diff
@@ -92,7 +92,7 @@ def sort_objects2(objects, id_getter, expr_getter, check_circular=True):
 
     def dependent_ids(obj):
         deps = dependencies(expr_getter(obj))
-        return [id_ if id_ in deps else None for id_ in known_ids]
+        return [id_ if id_ in deps else '' for id_ in known_ids]
 
     objects = sorted(objects, key=dependent_ids)
```

Comment:
I must admit I don't understand what's going on here – could you explain why we sort the `objects`, using a `list` as key? But that list has the same order as known_ids, which has no particular order at all, since it's a `set`, and `set`s are explicitly unordered.

Reply:
Wait is `known_ids` a set?